### PR TITLE
Product Tour - Welcome to FlowFuse

### LINKED ISSUE #2721: Interactive product tour
### Epic

_No response_

### Description

**As a: new user**

**I want to: be guided through the main features FlowFuse has to offer**

**So that: I can easily get to the "aha!" moment**

### Which customers would this be availble to

All Users, (CE)

### Acceptance Criteria

- [ ] criteria 1
- [ ] criteria 2
- [ ] criteria 3
- ...

### Have you provided an initial effort estimate for this issue?

I can not provide an initial effort estimate

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

- Add infrastructrue to render interactive product tours
   - ShepherdJS to run the tour logic
   - VueX store to track state on whether we should be showing tour or not
   - Tour composable that wraps ShepherdJS
   - Integrate PH analytics so we can track progression through the tours
- Adds the first "welcome" tour
   - Set to active if user arrives via the verify e-mail screen

## Related Issue(s)

Closes #2721 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 

https://github.com/user-attachments/assets/6bb40ee0-4955-4fde-9105-0a96001c1868

### CODE REVIEW

**Comment by joepavitt** [comment]
Example PH Funnel to track progress:

Screenshot 2024-08-15 at 15 39 45

**Review by Steve-Mcl — COMMENTED** [review]
Should we not be conditioning the tour or adapting to what is present in the team?

Also, is this intended for FFC or FFD also? 
Because while we do auto create an application and instance, FFD might not 🤷  

As an example, a user joining an existing team and signing in for the first time, when they access the existing team that has no instances or does have devices - it throws off the dialog.

image

image

**Comment by joepavitt** [comment]
Great point re: dedicated and joining existing teams... I'll adjust it accordingly. May need to be conditional (not sure how yet) on that last "Open Editor" step.

File: frontend/src/tours/tour-welcome.json
Comment (joepavitt) [inline frontend/src/tours/tour-welcome.json:1]: @gstout52 FYI - tour script is here if you want to review. @Steve-Mcl made great points that we need to make this host agnostic, so this covers all hosting options quite nicely, and even for new users coming into an existing team, full of resources.

**Review by Steve-Mcl — APPROVED** [review]
Agnostic language reads better. I feel this can support varying setups.

**Review by gstout52 — CHANGES_REQUESTED** [review]
Looks great! Two very small things:

Devices is currently:
<p><b>Devices</b> are Node-RED instances that are managed and deployed <i>remotely</i>.</p><p>Whether hardware in factories or a Raspberry Pi on your desk. FlowFuse helps you manage hundreds of remote deployments.</p>

I recommend combining the last two sentences: <p><b>Devices</b> are Node-RED instances that are managed and deployed <i>remotely</i>.</p><p>Whether hardware in factories or a Raspberry Pi on your desk, FlowFuse helps you manage hundreds of remote deployments.</p>

Open editor is currently:
<p><b>That's our quick introduction to FlowFuse</b></p><p>When you're ready, we recommend you open a Node-RED editor and start building your first flow.</p>

I recommend this slight change in language:
<p><b>That's our quick introduction to FlowFuse</b></p><p>When you're ready, we recommend that you open a Node-RED editor and start building your first flow.</p>

File: frontend/src/tours/tour-welcome.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}, {
+    "title": "Concept: Instances",
+    "text": "<p><b>Instances</b> refer to instances of Node-RED running at the <i>same</i> host as FlowFuse.</p>",
+    "attachTo": {
+        "element": "div[data-el=\"application-instance-item\"]",
+        "on": "bottom"
+    }
+}, {
+    "title": "Concept: Devices",
+    "text": "<p><b>Devices</b> are Node-RED instances that are managed and deployed <i>remotely</i>.</p><p>Whether hardware in factories or a Raspberry Pi on your desk. FlowFuse helps you manage hundreds of remote deployments.</p>",
```
Comment (joepavitt) [inline frontend/src/tours/tour-welcome.json:21]:
```suggestion
    "text": "<p><b>Devices</b> are Node-RED instances that are managed and deployed <i>remotely</i>.</p><p>Whether hardware in factories or a Raspberry Pi on your desk, FlowFuse helps you manage hundreds of remote deployments.</p>",
```

File: frontend/src/tours/tour-welcome.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}, {
+    "title": "Concept: Device Groups",
+    "text": "<p><b>Device Groups</b> are collections of <b>Devices</b> that can be used in <b>Pipelines</b> in order to deploy the same Node-RED flows out to multiple pieces of hardware in one-click.</p>",
+    "attachTo": {
+        "element": "a[data-nav=\"application-device-groups\"] ",
+        "on": "left"
+    }
+}, {
+    "title": "Open Node-RED Editor",
+    "text": "<p><b>That's our quick introduction to FlowFuse</b></p><p>When you're ready, we recommend you open a Node-RED editor and start building your first flow.</p>",
```
Comment (gstout52) [inline frontend/src/tours/tour-welcome.json:56]:
```suggestion
    "text": "<p><b>That's our quick introduction to FlowFuse.</b></p><p>When you're ready, we recommend that you open a Node-RED editor and start building your first flow.</p>",
```

**Review by gstout52 — APPROVED** [review]

**Pull request merged**